FirebaseDataConnect/Tests/Integration/IntegrationTests.swift

```diff
@@ -12,10 +12,218 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import Foundation
+import XCTest
 
+import FirebaseCore
 @testable import FirebaseDataConnect
 
-import XCTest
+final class IntegrationTests: XCTestCase {
+  class func setupFirebaseApp() {
+    if FirebaseApp.app() == nil {
+      let options = FirebaseOptions(googleAppID: "0:0000000000000:ios:0000000000000000",
+                                    gcmSenderID: "00000000000000000-00000000000-000000000")
+      options.projectID = "fdc-test"
+      FirebaseApp.configure(options: options)
+    }
+  }
+
+  private let setupQ = DispatchQueue(
+    label: "setupFDCIntegrationsTests",
+    autoreleaseFrequency: .workItem
+  )
+  private var _setupComplete = false
+  private var setupComplete: Bool {
+    set { setupQ.sync { _setupComplete = newValue } }
+    get { setupQ.sync { _setupComplete }}
+  }
+
+  override class func setUp() {
+    setupFirebaseApp()
+    DataConnect.kitchenSinkClient.useEmulator(port: 3628)
+  }
+
+  override func setUp(completion: @escaping ((any Error)?) -> Void) {
+    Task {
+      print("setupComplete \(self.setupComplete) \(self)")
+      guard !self.setupComplete else {
+        // emulator already configured with test project
+        print("Project already configured")
+        return
+      }
+
+      do {
+        try await DataConnect.kitchenSinkClient.configureProject()
+        self.setupComplete = true
+        print("setupComplete = true")
+        completion(nil)
+      } catch {
+        self.setupComplete = false
+        completion(error)
+      }
+    }
+  }
+
+  // test to confirm that we can assign an explicit UUID
+  func testSpecifiedUUID() async throws {
+    let specifiedUUID = UUID()
+    let result = try await DataConnect.kitchenSinkClient.createTestIdMutationRef(id: specifiedUUID)
+      .execute()
+    XCTAssertEqual(result.data.testId_insert.id, specifiedUUID)
+  }
+
+  // test for an auto generated UUID assignment
+  func testAutoId() async throws {
+    let result = try await DataConnect.kitchenSinkClient.createTestAutoIdMutationRef().execute()
+    _ = result.data.testAutoId_insert.id
+    // if we get till here - we have successfully got a UUID and decoded it. So test is successful
+    XCTAssert(true)
+  }
+
+  func testStandardScalar() async throws {
+    let standardScalarUUID = UUID()
+    let testText = "Hello Firebase World"
+    let testDecimal = Double.random(in: 10.0 ... 999.0)
+    let testInt = Int(Int32.random(in: Int32.min ... Int32.max))
+    // The following fails since server returns a different value.
+    // The value is outside the 32-bit range and GQL Int is 32-bits.
+    // let testInt = -6196243450739521536
```

Comment:
Is there a buganizer ticket tracking this issue? If so, please mention it here

Reply:
Yes there is one.

Reply:
Could you add the buganizer ticket to the comment in the code, for future code readers to get more info?

Reply:
Added